src/components/annotations/draw.js

```diff
@@ -579,21 +579,21 @@ function drawOne(gd, index, opt, value) {
                         ann.call(Lib.setTranslate, xcenter, ycenter);
 
                         update[annbase + '.x'] = xa ?
-                            (options.x + dx / xa._m) :
+                            xa.l2r(xa.p2l(xa.l2p(xa.r2l(options.x)) + dx)) :
                             ((arrowX + dx - gs.l) / gs.w);
                         update[annbase + '.y'] = ya ?
-                            (options.y + dy / ya._m) :
+                            ya.l2r(ya.p2l(ya.l2p(ya.r2l(options.y)) + dy)) :
```

Comment:
Have you tested shapes under `{ editable: true }`? @n-riesco  did add some jasmine test cases for them, but you can never be too sure.